app/src/main/java/com/example/plantsapp/data/RoomTasksRepository.kt

```diff
@@ -1,34 +1,42 @@
 package com.example.plantsapp.data
 
 import androidx.core.net.toUri
+import com.example.plantsapp.data.dao.RoomPlantWithTasksDao
+import com.example.plantsapp.data.entity.RoomTask
 import com.example.plantsapp.domain.model.Task
 import com.example.plantsapp.domain.repository.TasksRepository
 import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.map
 import java.util.Date
 
 class RoomTasksRepository(
-    private val plantsDao: RoomPlantsDao
+    private val plantsWithTasksDao: RoomPlantWithTasksDao
 ) : TasksRepository {
 
+    override suspend fun addTasks(tasks: List<Task>) {
+        tasks.forEach {
+            plantsWithTasksDao.insert(
+                RoomTask.from(it)
+            )
+        }
+    }
+
     override suspend fun getTasksForDate(date: Date): Flow<List<Task>> {
-        return plantsDao.getAll()
-            .map { plants ->
-                plants.filter {
-                    checkIfDateIsWithinInterval(
-                        date.time,
-                        it.creationDateMillis,
-                        it.wateringFrequencyDays
-                    )
-                }
-                    .map {
-                        Task(
-                            // TODO remove task action setting from repository
-                            WATERING_ACTION,
-                            it.name,
-                            it.plantPicture?.toUri()
+        return plantsWithTasksDao.getPlantsWithTasks()
+            .map { plantsWithTasks ->
+                plantsWithTasks.map { plantWithTasks ->
+                    plantWithTasks.tasks.filter {
+                        checkIfDateIsWithinInterval(
```

Comment:
1 I think you can move this to some extensions and rename - because it somehow hard to read and understand

Reply:
2 maybe improve this code somehow